src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,128 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class Exercise1 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortComparator() {
+        Person[] persons = getPersons();
+
+        Comparator<Person> comparator = new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return Integer.compare(o1.getAge(), o2.getAge());
+            }
+        };
+
+        Arrays.sort(persons, comparator);
+
+        assertArrayEquals(new Person[]{
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return Integer.compare(o1.getAge(), o2.getAge());
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                int i = o1.getLastName().compareTo(o2.getLastName());
+
+                if (i != 0) {
+                    return i;
+                }
+                return o1.getFirstName().compareTo(o2.getFirstName());
```

Comment:
It's better to name important variables with meaningful names. 
For example, not `i` but `lastNameComparing` or `leftToRightLastNameComparing` - depends on context.

In addition, don't you like ternary operator? :)
Lines 66-69 could be simplier: 
```Java
return i != 0 ? i : o1.get....;
```
Sometimes ternary operator makes code harder to read, I'm agree with you. It's your decision what you should use.

Reply:
Yes, thanks for your effort!:)